### PR TITLE
Cache by schema version

### PR DESCRIPTION
**Pull request by fantix**
Drop integer `dbver` and replace it with the UUID `schema_version`, so that we can persist the query cache under consistent keys in the next steps.

`dbver` was bumped on config changes but `schema_version` is only for user schema. We handle this difference by adding such config into the compile cache key / PG state key.

`get_session_config()` for cache key [should be optimized](https://github.com/edgedb/edgedb/issues/6705) to honor `affects_compilation`, this is fixed in a future persistent cache PR.

### CODE REVIEW

File: edb/server/dbview/dbview.pyx [hunk 1]
```diff
@@ -542,9 +537,15 @@ cdef class DatabaseConnectionView:
...
                 'no need to serialize state while in transaction')
 
-        dbver = self._db.dbver
+        ver = hash(
+            (
+                self._db.schema_version,
+                self.get_database_config(),
+                self.get_system_config(),
+            )
+        )
```
Comment (msullivan) [inline edb/server/dbview/dbview.pyx:546]: I don't think that the python `hash` function is reliable enough to use as a cache key

Comment (msullivan) [reply edb/server/dbview/dbview.pyx:546]: (I think I like the third option best?)

**Review by msullivan — APPROVED** [review]
This looks good, modulo the concerns I have about the changes to `_config_cache` invalidation

File: edb/server/dbview/dbview.pyx [hunk 2]
```diff
@@ -557,12 +558,12 @@ cdef class DatabaseConnectionView:
                 state.append({"name": sval.name, "value": jval, "type": kind})
 
         # Include the database version in the state so that we are forced
-        # to clear the config cache on dbver changes.
+        # to clear the config cache on changes.
         state.append(
-            {"name": '__dbver__', "value": dbver, "type": 'C'})
+            {"name": '__ver__', "value": ver, "type": 'C'})
```
Comment (elprans) [inline edb/server/dbview/dbview.pyx:563]: Where are we actually reading this?

Comment (msullivan) [reply edb/server/dbview/dbview.pyx:563]: https://github.com/edgedb/edgedb/blob/669ba34661ad04a605b12f745fa4b92122557570/edb/server/protocol/execute.pyx#L88-L91 and https://github.com/edgedb/edgedb/blob/669ba34661ad04a605b12f745fa4b92122557570/edb/server/protocol/execute.pyx#L253-L256 are the relevant places.

It doesn't get used directly, but changes in it are used to force a state sync

File: edb/server/dbview/dbview.pyx (hunk 1 above)
Comment (fantix) [reply edb/server/dbview/dbview.pyx:546]: In that case, I think it's easier if we keep the `dbver`, but only use schema_version for compile cache.

**Review by msullivan — APPROVED** [review]

**Pull request merged**